### PR TITLE
The `TodoTemplate` cleanup (#2343)

### LINKED ISSUE #2343: TodoTemplate project Cleanup
Currently, there are some misuses of the name `TodoTemplate` or `todoTemplate` in the projects of the TodoTemplate which creates some issues in the generated projects using the Project Template. it is required to change these names to other appropriate replacements.

### PR DESCRIPTION
**Pull request by msynk**
this closes #2343

### CODE REVIEW

File: src/Templates/TodoTemplate/Bit.TodoTemplate/App/TodoTemplate.App.csproj [hunk 1]
```diff
@@ -12,7 +12,7 @@
         <ApplicationTitle>TodoTemplate</ApplicationTitle>
 
         <!-- App Identifier -->
-        <ApplicationId>com.bitplatform.TodoTemplate</ApplicationId>
+        <ApplicationId>com.bitplatform.Todo.Template</ApplicationId>
```
Comment (ysmoradi) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/App/TodoTemplate.App.csproj:15]: Why? @msynk

Comment (msynk) [reply src/Templates/TodoTemplate/Bit.TodoTemplate/App/TodoTemplate.App.csproj:15]: if we don't change it, the `TodoTemplate` is going to be replaced with the target project name which is meaningless:
`com.bitplatform.testApp`

File: src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Services/Implementations/AuthenticationService.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+public class AuthenticationService : IAuthenticationService
+{
+    private readonly HttpClient _httpClient;
+
+    private readonly IJSRuntime _jsRuntime;
+
+    private readonly AppAuthenticationStateProvider _authenticationStateProvider;
+
+    public AuthenticationService(HttpClient httpClient, IJSRuntime jsRuntime,
```
Comment (ysmoradi) [inline src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Services/Implementations/AuthenticationService.cs:13]: Shall we use AutoInject here? @msynk

**Review by ysmoradi — DISMISSED** [review]

File: src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Services/Implementations/AuthenticationService.cs (hunk 2 above)
Comment (msynk) [reply src/Templates/TodoTemplate/Bit.TodoTemplate/Web/Services/Implementations/AuthenticationService.cs:13]: sure

**Review by ysmoradi — APPROVED** [review]

**Pull request merged**